iati/core/tests/test_rulesets.py

```diff
@@ -142,29 +142,26 @@ def test_rule_class_cannot_be_instantiated_directly_with_name(self):
 class TestRuleSubclasses(object):
     """A container for tests relating to all Rule subclasses."""
 
-    rule_constructors = list(map(iati.core.rulesets.locate_constructor_for_rule_type, iati.core.rulesets._VALID_RULE_TYPES))
-    """A list of constructors for the various types of Rule."""
+    @pytest.fixture(params=[
+        iati.core.rulesets.locate_constructor_for_rule_type(rule_type) for rule_type in iati.core.rulesets._VALID_RULE_TYPES
+    ])
+    def rule_constructor(self, request):
+        """List constructors for the various types of Rule."""
```

Comment:
A particular fixture is only a single thing at any given time. This docstring sounds slightly too this-is-an-array-like.